src/views/virtualmachines/tree/hooks/useContextMenuListener.tsx

```diff
@@ -0,0 +1,34 @@
+import { useCallback, useLayoutEffect, useState } from 'react';
+
+import { TreeViewDataItem } from '@patternfly/react-core';
+
+type UseContextMenuListenerType = (
+  treeData: TreeViewDataItem[],
+) => [triggerElement: HTMLElement | null, hide: () => void];
+
+const useContextMenuListener: UseContextMenuListenerType = (treeData) => {
```

Comment:
contextMenu is the event, but a better name will be right click?

Reply:
okay good